### PR TITLE
return only the current address book from state proof alpha REST API

### PR DESCRIPTION
**Pull request by xin-hedera**
Signed-off-by: Xin Li <xin.li@swirlds.com>

**Detailed description**:

- Update state proof alpha REST API to only return the address book at the time of the queried transaction
- Update test data and integration test spec to validate the change
- Update design doc

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/stateproof.js [hunk 1]
```diff
@@ -116,10 +117,18 @@ let getAddressBooksAndNodeAccountIdsByConsensusNs = async (consensusNs) => {
          ON ab.start_consensus_timestamp = abe.consensus_timestamp
        WHERE start_consensus_timestamp <= $1
          AND file_id = 102
-       GROUP BY start_consensus_timestamp
-       ORDER BY start_consensus_timestamp`;
+       GROUP BY start_consensus_timestamp`;
+  if (!onlyCurrent) {
```
Comment (steven-sheehy) [inline hedera-mirror-rest/stateproof.js:121]: Would be better to make this a configuration option so we don't have to undo this change in the future. Something like `hedera.mirror.rest.stateproof.addressBookHistory=false`.

Comment (xin-hedera) [reply hedera-mirror-rest/stateproof.js:121]: sure. just have a little concern that if we should expose such an option or not. I'll implement it.

File: hedera-mirror-rest/stateproof.js [hunk 2]
```diff
@@ -116,10 +117,18 @@ let getAddressBooksAndNodeAccountIdsByConsensusNs = async (consensusNs) => {
...
+      ORDER BY start_consensus_timestamp`;
+  } else {
+    sqlQuery += `
+      ORDER BY start_consensus_timestamp DESC
+      LIMIT 1`;
+  }
+
   if (logger.isTraceEnabled()) {
-    logger.trace(`getAddressBooksAndNodeAccountIDsByConsensusNs: ${sqlQuery}, ${JSON.stringify(sqlParams)}`);
+    logger.info(`getAddressBooksAndNodeAccountIDsByConsensusNs: ${sqlQuery}, ${JSON.stringify(sqlParams)}`);
```
Comment (xin-hedera) [inline hedera-mirror-rest/stateproof.js:131]: oops, this is by mistake. will revert back to logger.trace

File: hedera-mirror-rest/stateproof.js (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/stateproof.js]: added the option and it's not documented in configuration.md

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**